storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -308,6 +308,21 @@ public void updateWeakSubjectivityState(WeakSubjectivityUpdate weakSubjectivityU
     }
   }
 
+  @Override
+  public void storeFinalizedState(BeaconState state) {
+    if (state == null) {
+      return;
+    }
+
+    final Bytes32 blockRoot = spec.getBlockRootAtSlot(state, state.getSlot());
+    try (final FinalizedUpdaterCommon updater = getFinalizedUpdater()) {
+      updater.addFinalizedState(blockRoot, state);
+      updater.commit();
+    }
+  }
+
+  protected abstract FinalizedUpdaterCommon getFinalizedUpdater();
```

Comment:
I think you can use the existing `finalizedUpdater()` method rather than adding this. The `FinalizedUpdaterT` generic type it returns is guaranteed to be a subclass of `FinalizedUpdaterCommon`.